src/main/java/org/isf/menu/manager/UserSettingManager.java

```diff
@@ -51,32 +52,57 @@ public UserSetting newUserSetting(UserSetting userSetting) throws OHServiceExcep
 	 * Updates an existing {@link UserSetting} in the DB.
 	 *
 	 * @param userSetting - the {@link UserSetting} to insert
-	 * @return <code>{@link UserSetting}</code> if the userSetting has been updated,
-	 *         <code>null</code> otherwise.
+	 * @return {@link UserSetting} if the userSetting has been updated,
+	 *         {@code null} otherwise.
 	 * @throws OHServiceException
 	 */
 	public UserSetting updateUserSetting(UserSetting userSetting) throws OHServiceException {
 		return userSettingIoOperationRepository.save(userSetting);
 	}
-
+	
 	/**
 	 * Returns the list of {@link UserSetting}s of the specified userId.
 	 *
-	 * @param userId - the user Id
+	 * @param userName - the user name
 	 * @return the list of {@link UserSetting}s
 	 */
-	public List<UserSetting> getUserSetting(String userId) throws OHServiceException {
-		return userSettingIoOperationRepository.findAllByUSerID(userId);
+	public List<UserSetting> getUserSettingByUserName(String userName) throws OHServiceException {
+		return userSettingIoOperationRepository.findAllByUSerName(userName);
 	}
 
 	/**
 	 * Returns {@link UserSetting}s of the specified userId.
 	 *
-	 * @param userId     - the user Id
+	 * @param userName     - the user name
 	 * @param configName - the name of the user setting
-	 * @return <code>{@link UserSetting}</code> if the userSetting exist, <code>null</code> otherwise.
+	 * @return {@link UserSetting} if the userSetting exist, {@code null} otherwise.
+	 */
+	public UserSetting getUserSettingByUserNameConfigName(String userName, String configName) throws OHServiceException {
+		return userSettingIoOperationRepository.findByUserNameAndConfigName(userName, configName);
+	}
+	
+	/**
+	 * Returns {@link UserSetting}s of the specified userId.
+	 *
+	 * @param userName     - the user name
+	 * @return {@link UserSetting} if the userSetting exist, {@code null} otherwise.
+	 */
+	public UserSetting getUserSettingById(int userSettingId) throws OHServiceException {
+		Optional<UserSetting> us = userSettingIoOperationRepository.findById(userSettingId);
+		if (us.isPresent())
+			return us.get();
+		return null;
+	}
+
+	/**
+	 * Delete the {@link UserSetting}.
+	 * @param userSetting - the userSetting to delete
+	 * @return {@code true} if the {@link UserSetting} has been deleted, {@code false} otherwise.
+	 * @throws OHServiceException
 	 */
-	public UserSetting getUserSetting(String userId, String configName) throws OHServiceException {
-		return userSettingIoOperationRepository.findUserSettingByUSerID(userId, configName);
+	public Boolean deleteUserSetting(UserSetting userSetting) throws OHServiceException {
```

Comment:
Proper JPA crud routines are defined to return `void` not a value; worse yet here you specify an object instead of a primitive.   The body of the routine should be line 105.   See issue OP-1130 where I'm going through removing code like this.